api/src/main/java/org/terrakube/api/plugin/state/RemoteTfeService.java

```diff
@@ -929,41 +949,16 @@ RunsData getRun(int runId, String include) {
             return null;
     }
 
-    int calculateRunPositionInQueue(Job job) {
-        Optional<List<Job>> jobList = jobRepository.findAllByWorkspaceAndStatusNotInOrderByIdAsc(
-                job.getWorkspace(),
-                Arrays.asList(
-                        JobStatus.failed,
-                        JobStatus.completed,
-                        JobStatus.rejected,
-                        JobStatus.cancelled,
-                        JobStatus.approved));
-        int positionQueueInQueue = 0;
-        if (jobList.isPresent()) {
-            for (Job jobInQueue : jobList.get()) {
-                if (jobInQueue.getId() == job.getId()) {
-                    break;
-                }
-                positionQueueInQueue += 1;
-            }
-        }
-        log.info("Position in queue {} for workspace {}", positionQueueInQueue, job.getWorkspace().getName());
-        return positionQueueInQueue;
-    }
-
     RunsDataList getRunsQueue(String organizationName) {
         RunsDataList runsDataList = new RunsDataList();
-        runsDataList.setData(new ArrayList());
-        Organization organization = organizationRepository.getOrganizationByName(organizationName);
-        List<Job> jobList = jobRepository.findAllByOrganizationAndStatusNotInOrderByIdAsc(
-                organization,
+        runsDataList.setData(new ArrayList<RunsModel>());
+        List<Job> jobList = jobRepository.findAllByStatusInOrderByIdAsc(
                 Arrays.asList(
-                        JobStatus.failed,
-                        JobStatus.completed,
-                        JobStatus.rejected,
-                        JobStatus.cancelled,
-                        JobStatus.approved));
-        int runQueue = 0;
+                        JobStatus.pending,
+                        JobStatus.running,
+                        JobStatus.queue));
```

Comment:
I think you need to add JobStatus.waitingForApproval here

Reply:
Don't think you need it here because this function is used to calculate the capacity of the runners to see when the Job can be queued.